code/modules/materials/Mat_MaterialProcs.dm

```diff
@@ -400,24 +400,32 @@ triggerOnEntered(var/atom/owner, var/atom/entering)
 
 		if(agent_b && temp > 500 && air.toxins > MINIMUM_REACT_QUANTITY )
 			var/datum/gas/oxygen_agent_b/trace_gas = payload.get_or_add_trace_gas_by_type(/datum/gas/oxygen_agent_b)
-			payload.temperature = T0C // Greatly reduce temperature to simulate an endothermic reaction
-
-			// Itr 1: 0.125 Agent B, 10 Oxy
+			if(temp>10000)
+				payload.temperature = temp -= 10000 // Greatly reduce temperature to simulate an endothermic reaction
```

Comment:
You don't need to assign this to temp.

Is there anything significant about the change of heat by 10000?